core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaRecordDeserializer.java

```diff
@@ -0,0 +1,72 @@
+package org.zalando.nakadi.repository.kafka;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.dataformat.avro.AvroMapper;
+import org.zalando.nakadi.domain.EnvelopeHolder;
+import org.zalando.nakadi.domain.NakadiRecord;
+import org.zalando.nakadi.service.AvroSchema;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+public class KafkaRecordDeserializer implements RecordDeserializer {
+
+    private final AvroSchema schemas;
+    private final AvroMapper avroMapper;
+    private final ObjectMapper objectMapper;
+
+    public KafkaRecordDeserializer(final AvroSchema schemas) {
+        this.schemas = schemas;
+        this.avroMapper = schemas.getAvroMapper();
+        this.objectMapper = schemas.getObjectMapper();
+    }
+
+    public byte[] deserialize(final byte[] eventFormat, final byte[] data) {
+        if (eventFormat == null) {
+            // JSON
+            return data;
+        }
+
+        if (Arrays.equals(eventFormat, NakadiRecord.Format.AVRO.getFormat())) {
+            try {
+                final EnvelopeHolder envelop = EnvelopeHolder.fromBytes(data);
+                if (envelop.getMetadataVersion() != AvroSchema.METADATA_VERSION) {
+                    throw new RuntimeException(String.format(
+                            "metadata version is not supported: `%d`",
```

Comment:
Nitpick mode: seems like in Java world, first capital letter for an exception message is customary, so:
```suggestion
                            "Metadata version is not supported: `%d`",
```
And in other places as well.

Reply:
I prefer to start from the lower case, it is nicer in the logs